src/shogun/mathematics/linalg/LinalgBackendViennacl.h

```diff
@@ -115,6 +132,14 @@ class LinalgBackendViennaCL : public LinalgBackendGPUBase
 		return viennacl::linalg::inner_prod(a_gpu->data(a.size()), b_gpu->data(b.size()));
 	}
 
+	/** ViennaCL vector sum method */
+	template <typename T>
+	T sum_impl(const SGVector<T>& vec) const
+	{
+		GPUMemoryViennaCL<T>* vec_gpu=static_cast<GPUMemoryViennaCL<T>*>(vec.gpu_vector.get());
```

Comment:
as said, it would be good to make a private helper method for this and use it everywhere